app/models/name.rb

```diff
@@ -408,33 +457,181 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
-
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
+  scope :deprecated, -> { where(deprecated: true) }
   scope :not_deprecated, -> { where(deprecated: false) }
+  scope :with_description, -> { where.not(description_id: nil) }
+  scope :without_description, -> { where(description_id: nil) }
+  # Names without descriptions, order by most frequently used
+  scope :description_needed, lambda {
+    without_description.joins(:observations).
+      group(:name_id).order(Arel.star.count.desc)
+  }
+  scope :description_includes,
+        lambda { |text|
+          joins(:descriptions).
+            where(NameDescription[:gen_desc].matches("%#{text}%")).
+            or(where(NameDescription[:diag_desc].matches("%#{text}%"))).
+            or(where(NameDescription[:distribution].matches("%#{text}%"))).
+            or(where(NameDescription[:habitat].matches("%#{text}%"))).
+            or(where(NameDescription[:look_alikes].matches("%#{text}%"))).
+            or(where(NameDescription[:notes].matches("%#{text}%"))).
+            or(where(NameDescription[:refs].matches("%#{text}%")))
+        }
+  scope :with_description_in_project,
```

Comment:
@nimmolo -- is my understanding correct: 
`Name.with_description_in_project(x)` means Names that both have a Description and are in the given Project.

Reply:
Your understanding of projects is probably better than mine, and if you can think of a better name please do change it.

Reply:
@JoeCohen I'm about to leave but i had to look it up. 

I remember now, I was going through the [app/classes/query](https://github.com/MushroomObserver/mushroom-observer/blob/master/app/classes/query) folder and pulling queries for observations, names and locations. It seems I must have meant to write the entirely different `[Name.with_observations_in_project(x)](https://github.com/MushroomObserver/mushroom-observer/blob/master/app/classes/query/name_with_observations_for_project.rb)`! Sorry not to have investigated sooner. 

In any case, this directory, in combination with PatternSearch is a great starting point for finding further scopes.

Reply:
It seems i'm missing `Name.with_descriptions_by_author` and `Name.with_descriptions_by_editor` - if you want to add those, please do. 

Maybe they should be `with_description_authored_by` and `with_description_edited_by`, or maybe the existing ones should be `with_description_created_by` should be `with_descriptions_by_user`? No preference here. I don't know if I changed it, or picked up the usage in PatternSearch. I really have to finish packing! 😌

Reply:
Maybe those are all different scopes, though...

Reply:
@nimmolo: Thanks so much for looking it up.
It's actually a reasonable search. I'll leave it there for now, but anyone can rip it out.
I'll add the missing ones.
Have a great trip, and stay safe and healthy.